src/packages/server/projects/control/util.ts

```diff
@@ -113,6 +113,14 @@ export async function launchProjectDaemon(env, uid?: number): Promise<void> {
       uid,
       gid: uid,
     });
+    child.stdout.on('data', (data) => {
```

Comment:
Please only do this if the logging is actually enabled.  E.g., `stdout ${data.toString("utf8")}` could take a lot of time to do nothing if logging isn't even enabled.

Reply:
I removed additional logging. Actually, regular stdout here consists of only a couple of lines before running the project daemon, but it's okay. I don't want to check the logging level here because all the other debug calls around it don't do it.